internal/nri/nri.go

```diff
@@ -169,6 +172,26 @@ func (l *local) RunPodSandbox(ctx context.Context, pod PodSandbox) error {
 	return err
 }
 
+func (l *local) UpdatePodSandbox(ctx context.Context, pod PodSandbox) error {
+	if !l.IsEnabled() {
+		return nil
+	}
+
+	l.Lock()
+	defer l.Unlock()
+
+	podNri := podSandboxToNRI(pod)
+	request := &nri.UpdatePodSandboxRequest{
+		Pod:                    podNri,
+		OverheadLinuxResources: podNri.Linux.GetPodOverhead(),
+		LinuxResources:         podNri.Linux.GetPodResources(),
+	}
+
+	_, err := l.nri.UpdatePodSandbox(ctx, request)
+
+	return err
+}
```

Comment:
I think we should change this, adding extra arguments for the updated `overhead` and `resources`, then pass those on to NRI in the request. Otherwise NRI gets is the old pod with `nil` updates. So something like this:

```
func (l *local) UpdatePodSandbox(ctx context.Context, pod PodSandbox, overhead, resources *nri.LinuxResources) error {
	if !l.IsEnabled() {
		return nil
	}

	l.Lock()
	defer l.Unlock()

	podNri := podSandboxToNRI(pod)
	request := &nri.UpdatePodSandboxRequest{
		Pod:                    podNri,
		OverheadLinuxResources: overhead,
		LinuxResources:         resources,
	}

	_, err := l.nri.UpdatePodSandbox(ctx, request)

	return err
}
```